### PR TITLE
:seedling: Add jobs for CAPI release 1.9 and remove 1.6 jobs

### PR DESCRIPTION
**Pull request by cahillsf**
adds new jobs for `release-1.9` branch and drops jobs for `release-1.6` per release docs: https://github.com/kubernetes-sigs/cluster-api/tree/main/docs/release/role-handbooks/ci-signal#setup-jobs-and-dashboards-for-a-new-release-branch

### CODE REVIEW

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-prowjob-gen.yaml [hunk 1]
```diff
@@ -107,6 +105,8 @@ prow_ignored:
         to: "1.28"
       - from: "1.28"
         to: "1.29"
+      - from: "1.29"
+        to: "1.30"
```
Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-prowjob-gen.yaml:109]: one nit: we can drop `"1.23"` below at versions.

File: config/jobs/kubernetes-sigs/cluster-api/cluster-api-prowjob-gen.yaml [hunk 2]
```diff
@@ -119,11 +119,6 @@ prow_ignored:
     template: 'cluster-api-{{ ReplaceAll .branch "." "-" }}-presubmits.yaml'
 
   # Versions provides kubernetes minor version to key-value pairs mapping.
```
Comment (chrischdi) [inline config/jobs/kubernetes-sigs/cluster-api/cluster-api-prowjob-gen.yaml:121]:
```suggestion
  # Versions provides kubernetes minor version to key-value pairs mapping.
  versions:
```

**Pull request merged**